internal/permissions/permission_store.go

```diff
@@ -30,110 +27,176 @@ type Store interface {
 	ListReadPermissionsByLexicon(owner string) (map[string][]string, error)
 }
 
-type casbinStore struct {
-	enforcer *casbin.Enforcer
-	adapter  persist.Adapter
+type sqliteStore struct {
+	db *gorm.DB
 }
 
-//go:embed model.conf
-var modelStr string
+var _ Store = (*sqliteStore)(nil)
 
-func NewStore(adapter persist.Adapter, autoSave bool) (Store, error) {
-	m, err := model.NewModelFromString(modelStr)
-	if err != nil {
-		return nil, err
-	}
-	enforcer, err := casbin.NewEnforcer(m, adapter)
+// Permission represents a permission entry in the database
+type Permission struct {
+	gorm.Model
+	Grantee string `gorm:"not null;index:idx_permissions_grantee_owner,priority:1;uniqueIndex:idx_grantee_owner_object"`
+	Owner   string `gorm:"not null;index:idx_permissions_owner;index:idx_permissions_grantee_owner,priority:2;uniqueIndex:idx_grantee_owner_object"`
```

Comment:
what does `priority` do? is that for sort ordering?

Reply:
also where are the `index`es defined / created? because these lines don't define the indexes themselves.

Reply:
Yeah priority is the primary key order. Gorm handles all the table and index creation in AutoMigrate